client/landing/stepper/declarative-flow/internals/components/step-route/index.tsx

```diff
@@ -11,9 +28,92 @@ type StepRouteProps = {
 	renderStep: ( step: StepperStep ) => JSX.Element | null;
 };
 
+//TODO: Check we can move RenderStep function to here and remove the renderStep prop
 const StepRoute = ( { step, flow, showWooLogo, renderStep }: StepRouteProps ) => {
+	const userIsLoggedIn = useSelector( isUserLoggedIn );
+	const loginUrl = useLoginUrlForFlow( { flow } );
 	const stepContent = renderStep( step );
+	const { site, siteSlugOrId } = useSiteData();
 
+	// Ensure that the selected site is fetched, if available. This is used for event tracking purposes.
+	// See https://github.com/Automattic/wp-calypso/pull/82981.
+
+	// TODO: Check if it is duplicated with the useSiteData hook
+	const selectedSite = useSelector( ( state ) => site && getSite( state, siteSlugOrId ) );
+	const isRequestingSelectedSite = useSelector(
+		( state ) => site && isRequestingSite( state, siteSlugOrId )
+	);
+	const intent = useIntent();
+	const design = useSelectedDesign();
```

Comment:
Out of scope but I really dislike Stepper being aware of designs. Tracks should probably support a middleware to allow deeper code to intercept events, and add extra information.

Reply:
💯 Fully agreed! But I didn't want to break the Tracks events as part of trying to fix them! 🙃